src/cljam/io/bam/decoder.clj

```diff
@@ -129,9 +144,13 @@
          seq         (decode-seq (lsb/read-bytes buffer (quot (inc l-seq) 2)) l-seq)
          qual        (decode-qual (lsb/read-bytes buffer l-seq))
          rest        (lsb/read-bytes buffer (options-size (alength ^bytes (:data block)) l-read-name n-cigar-op l-seq))
-         options     (decode-options rest)]
+         options     (decode-options rest)
+         [cigar* options*]
+         (if (and (cigar/placeholder? cigar-bytes) (some :CG options))
+           [(get-cigar-from-options options) (remove :CG options)]
```

Comment:
`if-let` can be used to gather operations on the `options` sequence itself in one place, and separate them from operations on the content of the element.

```suggestion
         (if-let [cg (and (cigar/placeholder? cigar-bytes)
                          (:value (some :CG options)))]
           [(B-I-type-cigar-str->cigar-str cg) (remove :CG options)]
```